__tests__/hooks/usePrefersReducedMotion.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { usePrefersReducedMotion } from '../../src/hooks/usePrefersReducedMotion';
+
+describe('usePrefersReducedMotion', () => {
+  it.each([
+    { query: 'no-preference', prefersReducedMotion: false },
+    { query: 'reduce', prefersReducedMotion: true },
+  ])(
+    'should return a boolean indicating whether or not the user prefers reduced motion',
+    ({ query, prefersReducedMotion }) => {
+      window.matchMedia = jest.fn().mockImplementation((q) => ({
```

Comment:
did you run into the same issue where `jsdom` doesn't actually emulate this property yet so you can't do a proper `jest.spy` or similar style of mock implementation? not sure if its worth calling out the limitation. 

Also should we have an `afterAll` that resets this to `window.matchMedia = undefined` or anything to avoid possible pollution/bugs elsewhere. The potential feels low, but seems like it'd be good to watch our butts so to speak

Reply:
Yep, exactly. Jest has a good guide on `window.matchMedia` specifically, and this is adapted from that: https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

Good shout on the `afterAll`, I'll add that in.